packages/frontapp/src/app/components/BooksByCategory/index.tsx

```diff
@@ -1,42 +1,127 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useGetAllMaterialsQuery } from '@mimir/apollo-client';
 import { GoBack, ButtonWrapper } from '../../pages/BookPreview';
 import { ReactComponent as ArrowBack } from '../../../assets/ArrowUp2.svg';
 import { useParams, useNavigate } from 'react-router-dom';
 import BookCard from '../BookCard';
 import { WrapperList } from '../ListBooks';
+import { useSearchParams } from 'react-router-dom';
+import { Material } from '@mimir/apollo-client';
+
+type IMaterial =
+  | null
+  | undefined
+  | Pick<
+      Material,
+      'id' | 'title' | 'author' | 'type' | 'picture' | 'created_at' | 'category'
+    >;
```

Comment:
If you need to select a lot of properties, isn't it easier to specify those that you don't need?
Use Omit instead